src/librustc/hir/intravisit.rs

```diff
@@ -731,17 +736,17 @@ pub fn walk_ty_param_bound<'v, V: Visitor<'v>>(visitor: &mut V, bound: &'v TyPar
 }
 
 pub fn walk_generic_param<'v, V: Visitor<'v>>(visitor: &mut V, param: &'v GenericParam) {
-    match *param {
-        GenericParam::Lifetime(ref ld) => {
-            visitor.visit_lifetime(&ld.lifetime);
-            walk_list!(visitor, visit_lifetime, &ld.bounds);
-        }
-        GenericParam::Type(ref ty_param) => {
-            visitor.visit_id(ty_param.id);
-            visitor.visit_name(ty_param.span, ty_param.name);
-            walk_list!(visitor, visit_ty_param_bound, &ty_param.bounds);
-            walk_list!(visitor, visit_ty, &ty_param.default);
-            walk_list!(visitor, visit_attribute, ty_param.attrs.iter());
+    match param.kind {
+        GenericParamKind::Lifetime { ref bounds, ref lifetime_deprecated, .. } => {
+            visitor.visit_lifetime(lifetime_deprecated);
```

Comment:
This is the only place the `lifetime` field is used for `hir::GenericParamKind::Lifetime`. It'd be nice if I could get rid of it completely.